### PR TITLE
[Formatter] Add several node implementations

### PR DESCRIPTION
**Pull request by IrushiL**
## Purpose
**Adds the following node implementations:**

NilLiteralNode, StartActionNode, FlushActionNode, FailStatementNode, ContinueStatementNode, TrapExpressionNode, LetExpressionNode, LetVariableDeclarationNode, FromClauseNode, WhereClauseNode, LetClauseNode, QueryPipelineNode, WildcardBindingPatternNode, AsyncSendActionNode, SyncSendActionNode, ReceiveActionNode, ReceiveFieldsNode, WaitActionNode, QueryActionNode, TransactionStatementNode, RollbackStatementNode, RetryStatementNode, CommitActionNode, LimitClauseNode, JoinClauseNode, BreakStatementNode, XMLNamespaceDeclarationNode, ModuleXMLNamespaceDeclarationNode, XMLSimpleNameNode, XMLEmptyElementNode, XMLTextNode, XMLComment, XMLProcessingInstruction, LocalTypeDefinitionStatementNode, ForkStatementNode, UnaryExpressionNode, DefaultableParameterNode, AnnotationDeclarationNode, KeySpecifierNode, StreamTypeDescriptorNode, StreamTypeParamsNode, InterpolationNode, MethodDeclarationNode, DoubleGTTokenNode, TrippleGTTokenNode, OrderKeyNode

Partially Fixes #26052

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java [hunk 1]
```diff
@@ -1328,270 +1452,1389 @@ public ListBindingPatternNode transform(ListBindingPatternNode listBindingPatter
...
+
+    @Override
+    public OrderByClauseNode transform(OrderByClauseNode orderByClauseNode) {
+
+        return super.transform(orderByClauseNode);
+    }
+
+    @Override
+    public OrderKeyNode transform(OrderKeyNode orderKeyNode) {
+        ExpressionNode expression = formatNode(orderKeyNode.expression(), 1, 0);
```
Comment (SupunS) [inline misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java:2812]: I think the trailing WS/NL should change depending on the existence of the `orderDirection`

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java [hunk 2]
```diff
@@ -1328,270 +1452,1389 @@ public ListBindingPatternNode transform(ListBindingPatternNode listBindingPatter
...
+    public AnnotationDeclarationNode transform(AnnotationDeclarationNode annotationDeclarationNode) {
+        if (annotationDeclarationNode.metadata().isPresent()) {
+            MetadataNode metadata = formatNode(annotationDeclarationNode.metadata().get(), 0, 1);
+            annotationDeclarationNode = annotationDeclarationNode.modify().withMetadata(metadata).apply();
+        }
+
+        Token visibilityQualifier = formatToken(annotationDeclarationNode.visibilityQualifier(), 1, 0);
+        Token constKeyword = formatToken(annotationDeclarationNode.constKeyword(), 1, 0);
+        Token annotationKeyword = formatToken(annotationDeclarationNode.annotationKeyword(), 1, 0);
+        Node typeDescriptor = formatNode(annotationDeclarationNode.typeDescriptor(), 1, 0);
```
Comment (SupunS) [inline misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java:2180]: This also seems to be a bug in syntax-tree. Both `constKeyword` and `typeDescriptor` needs to be optional. Lets create an issue

**Review by SupunS — APPROVED** [review]

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java [hunk 3]
```diff
@@ -1328,270 +1452,1389 @@ public ListBindingPatternNode transform(ListBindingPatternNode listBindingPatter
...
+                .apply();
+    }
+
+    @Override
+    public EnumMemberNode transform(EnumMemberNode enumMemberNode) {
+        if (enumMemberNode.metadata().isPresent()) {
+            MetadataNode metadata = formatNode(enumMemberNode.metadata().get(), 0, 1);
+            enumMemberNode = enumMemberNode.modify().withMetadata(metadata).apply();
+        }
+        IdentifierToken identifier;
```
[the remark this answers is not in the document]
Comment (IrushiL) [reply misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java:1644]: This is modified in the if path as well.

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java [hunk 4]
```diff
@@ -1328,270 +1452,1389 @@ public ListBindingPatternNode transform(ListBindingPatternNode listBindingPatter
...
+                .withSemicolonToken(semicolonToken)
+                .apply();
+    }
+
+    @Override
+    public ModuleXMLNamespaceDeclarationNode transform(
+            ModuleXMLNamespaceDeclarationNode moduleXMLNamespaceDeclarationNode) {
+        Token xmlnsKeyword = formatToken(moduleXMLNamespaceDeclarationNode.xmlnsKeyword(), 1, 0);
+        ExpressionNode namespaceuri = formatNode(moduleXMLNamespaceDeclarationNode.namespaceuri(), 1, 0);
+        Token asKeyword = formatToken(moduleXMLNamespaceDeclarationNode.asKeyword(), 1, 0);
```
[the remark this answers is not in the document]
Comment (IrushiL) [reply misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java:1819]: https://github.com/ballerina-platform/ballerina-lang/issues/26328

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java (hunk 2 above)
Comment (IrushiL) [reply misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java:2180]: https://github.com/ballerina-platform/ballerina-lang/issues/26328

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java [hunk 5]
```diff
@@ -1328,270 +1452,1389 @@ public ListBindingPatternNode transform(ListBindingPatternNode listBindingPatter
...
+                .withName(name)
+                .withAttributes(attributes)
+                .withSlashToken(slashToken)
+                .withGetToken(getToken)
+                .apply();
+    }
+
+    @Override
+    public XMLTextNode transform(XMLTextNode xMLTextNode) {
+        Token content = formatToken(xMLTextNode.content(), this.trailingWS, this.trailingNL);
```
[the remark this answers is not in the document]
Comment (IrushiL) [reply misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java:1897]: We ideally shouldn't alter the value of an XML tag right?

**Pull request merged**